### PR TITLE
Add `lint` script, run it on Travis, prettify source code

### PR DESCRIPTION
**Pull request by josephfrazier**
This helps ensure that the source code for Prettier follows its own
formatting rules.

### CODE REVIEW

File: scripts/lint.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+if ! ./bin/prettier.js --list-different index.js src/*.js bin/*.js; then
```
Comment (vjeux) [inline scripts/lint.sh]: it's annoying that we have to list those paths in two different places. Do you think you could come up with a way to share the paths?

**Comment by vjeux** [comment]
I held up adding a lint rule for it because we were changing the format all the time, but now that it's way more stable, it's a good time :)

Could you rebase and i'll merge it in, thanks!

**Comment by lydell** [comment]
Won't it be annoying to always use the latest bleeding edge Prettier for linting? Feels like that will cause unnecessary diffs and merge conflicts to every PR. I think that installing Prettier for npm would be better.

My suggestion would be to add ESLint with eslint-plugin-prettier and a few of the most helpful ESLint rules such as [no-undef](http://eslint.org/docs/rules/no-undef). Then `eslint .` would lint, and `eslint . --fix` would format. Feels more cross platform than a bash script as well, and solves the problem of duplicating the globs. If you want I'd happily make a PR for it :)

**Comment by vjeux** [comment]
@lydell I really like your suggestion of using the latest published version. Same for rules like no-undef :)

If you send a PR for it, I'll merge it :)

**Comment by josephfrazier** [comment]
Great point @lydell, I just updated this PR to use `eslint-plugin-prettier` instead.

File: package.json [hunk 2]
```diff
@@ -23,8 +23,11 @@
   },
   "devDependencies": {
     "diff": "3.2.0",
+    "eslint": "^3.19.0",
+    "eslint-plugin-prettier": "^2.1.1",
```
Comment (vjeux) [inline package.json]: @zertosh :)

**Pull request merged**